examples/mnist_00.md

```diff
@@ -0,0 +1,242 @@
+    gamma: float = 0.7 # REMOVE THIS SINCE IT IS NOW WITHIN `AdadeltaConf` BELOW
+    no_cuda: bool = False
+    dry_run: bool = False
+    seed: int = 1
+    log_interval: int 
+    save_model: bool = False
+    adadelta: Any = AdadeltaConf()
+    steplr: Any = StepLRConf(step_size=1)
+```
+This works, but can feel a bit flat and disorganized (much like `argparse` args can be). Note, we also sacrifice `help` strings. Don't worry, we will remedy both of these concerns down the road.
+
+Now our `argparse` args are at the same level as our optimizer and scheduler configs. We will remove `lr` and `gamma` since they are already present within the optimizer config `AdadeltaConf`.
+***
+### Dropping into `main()`
+Now that we've defined all of our configs, we just need to let Hydra create our `cfg` object at runtime and make sure the `cfg` is plumbed to any object we want it to configure.
+```python
+@hydra.main(config_name='config')
+def main(cfg):
+    print(cfg.pretty())
+    ...
+```
+The single idea here is that `@hydra.main` looks for a config in the `ConfigStore` instance, `cs` named "`config`". It finds `MNISTConf` (our top level conf) and populates `cfg` inside `main()` with the entire structured config including our optimizer config, `cfg.adadelta` and our scheduler config, `cfg.steplr`.
+
+Instrumenting `main()` is simple. Anywhere we find `args`, replace this with `cfg` since we put all of the `argparse` arguments at the top level. For example, `args.batch_size` becomes `cfg.batch_size`:
+```python
+# the first few lines of main
+    ...
+    use_cuda = not cfg.no_cuda and torch.cuda.is_available() # DIFF args.no_cuda
+        torch.manual_seed(cfg.seed) # DIFF args.seed
+        device = torch.device("cuda" if use_cuda else "cpu")
+
+        train_kwargs = {'batch_size': cfg.batch_size} # DIFF args.batch_size
+        test_kwargs = {'batch_size': cfg.test_batch_size} # DIFF args.test_batch_size
+    ...
+```
+
+
+### Instantiating the optimizer and scheduler
+Still inside `main()`, we want to draw attention to two slightly special cases before moving on. Both the `optimizer` and `scheduler` are instantiated manually by specifying each argument with its `cfg` equivalent. Note that since these are nested fields, each of these parameters is two levels down e.g. `lr=args.learning_rate` becomes `lr=cfg.adadelta.lr`.
+
+```python
+ optimizer = Adadelta(lr=cfg.adadelta.lr, #DIFF lr=args.learning_rate
+                         rho=cfg.adadelta.rho,
+                         eps=cfg.adadelta.eps, 
+                         weight_decay=cfg.adadelta.weight_decay,
+                         params=model.parameters()
+ ```               
+In the case of the `optimizer`, one argument is not a part of our config -- `params`. If it wasn't obvious, this needs to be passed from the initialized `Net()` model. In the config schema that initialized `cfg.adadelta`, `params` is default to `MISSING`. The same is true of the `optimizer` field in `StepLRConf`.

```

Comment:
You have some Windows newlines here :)

Reply:
oh dear. not windows.

Haha, I wrote this in an online markdown editor because I was lazy. I guess they do some tomfoolery.